dotnet/src/Connectors/Connectors.AI.HuggingFace/TextCompletion/HuggingFaceTextCompletion.cs

```diff
@@ -120,7 +120,7 @@ private async Task<IReadOnlyList<ITextStreamingResult>> ExecuteGetCompletionsAsy
 
             return completionResponse.ConvertAll(c => new TextCompletionStreamingResult(c));
         }
-        catch (Exception e) when (e is not AIException && !e.IsCriticalException())
+        catch (Exception e) when (e is not AIException)
```

Comment:
MSIL for catch + when is way more costly than using the below. Apply the same for the others.

```suggestion
        catch (AIException aiex) 
        { 
            throw; 
        }
        catch (Exception ex)
```